users_ldap_push/models/res_users.py

```diff
@@ -0,0 +1,161 @@
+        'LDAP user', compute='_compute_is_ldap_user', default=True)
+
+    @api.model
+    @api.returns('self', lambda record: record.id)
+    def create(self, values):
+        result = super(ResUsers, self).create(values)
+        result.push_to_ldap(values)
+        return result
+
+    @api.multi
+    def write(self, values):
+        result = super(ResUsers, self).write(values)
+        self.push_to_ldap(values)
+        return result
+
+    @api.multi
+    def _push_to_ldap_possible(self, values):
+        return bool(self._get_ldap_configuration())
+
+    @api.multi
+    def _get_ldap_configuration(self):
+        self.ensure_one()
+        return self.sudo().company_id.ldaps.filtered('create_ldap_entry')[:1]
+
+    @api.multi
+    def _get_ldap_values(self, values):
+        self.ensure_one()
+        conf = self._get_ldap_configuration()
+        result = {}
+        for mapping in conf.create_ldap_entry_field_mappings:
+            field_name = mapping.field_id.name
+            if field_name not in values or not values[field_name]:
+                continue
+            result[mapping.attribute] = [str(values[field_name])]
+        if result:
+            result['objectClass'] = conf.create_ldap_entry_objectclass\
+                .encode('utf-8').split(',')
+        return result
+
+    @api.multi
+    def _get_ldap_dn(self, values):
+        self.ensure_one()
+        conf = self._get_ldap_configuration()
+        dn = conf.create_ldap_entry_field_mappings.filtered('use_for_dn')
+        assert dn, 'No DN attribute mapping given!'
+        assert self[dn.field_id.name], 'DN attribute empty!'
+        return '%s=%s,%s' % (
+            dn.attribute,
+            ldap.dn.escape_dn_chars(self[dn.field_id.name].encode('utf-8')),
+            conf.create_ldap_entry_base or conf.ldap_base)
+
+    @api.multi
+    def push_to_ldap(self, values):
+        for this in self:
+            if not values.get('is_ldap_user') and not this.is_ldap_user:
+                continue
+            if not this._push_to_ldap_possible(values):
+                continue
+            ldap_values = this._get_ldap_values(values)
+            if not ldap_values:
+                continue
+            ldap_configuration = this._get_ldap_configuration()
+            ldap_connection = ldap_configuration.connect(
+                ldap_configuration.read()[0])
+            ldap_connection.simple_bind_s(
+                (ldap_configuration.ldap_binddn or '').encode('utf-8'),
+                (ldap_configuration.ldap_password or '').encode('utf-8'))
+
+            if not this.ldap_entry_dn:
+                dn = this._get_ldap_dn(values)
+                try:
+                    ldap_connection.add_s(
+                        dn,
+                        ldap.modlist.addModlist(ldap_values))
+                except ldap.LDAPError, e:
+                    ldap_connection.unbind_s()
+                    raise exceptions.Warning(_('Error'), e.message)
```

Comment:
would we want to log something, in case the action is triggered from an ir.cron for instance?